appium/webdriver/webdriver.py

```diff
@@ -200,32 +213,34 @@ class WebDriver(
     Sms,
     SystemBars,
 ):
-    def __init__(
+    def __init__(  # noqa: PLR0913
         self,
         command_executor: Union[str, AppiumConnection] = 'http://127.0.0.1:4444/wd/hub',
         keep_alive: bool = True,
         direct_connection: bool = True,
         extensions: Optional[List['WebDriver']] = None,
         strict_ssl: bool = True,
         options: Union[AppiumOptions, List[AppiumOptions], None] = None,
+        client_config: Optional[ClientConfig] = None,
     ):
-        if strict_ssl is False:
-            # noinspection PyPackageRequirements
-            import urllib3
-
-            # noinspection PyPackageRequirements
-            import urllib3.exceptions
-
-            # noinspection PyUnresolvedReferences
-            AppiumConnection.set_certificate_bundle_path(None)
-            urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
-
         if isinstance(command_executor, str):
-            command_executor = AppiumConnection(command_executor, keep_alive=keep_alive)
+            client_config = client_config or ClientConfig(
+                remote_server_addr=command_executor, keep_alive=keep_alive, ignore_certificates=not strict_ssl
+            )
+            client_config.remote_server_addr = command_executor
+            command_executor = AppiumConnection(remote_server_addr=command_executor, client_config=client_config)
+        elif isinstance(command_executor, AppiumConnection) and strict_ssl is False:
+            logger.warning(
+                "Please set 'ignore_certificates' in the given 'appium.webdriver.appium_connection.AppiumConnection' or "
+                "'selenium.webdriver.remote.client_config.ClientConfig' instead. Ignoring."
+            )
 
         super().__init__(
             command_executor=command_executor,
             options=options,
+            locator_converter=AppiumLocatorConverter(),
```

Comment:
does this cache AppiumLocatorConverter instance or a new one is created every time a new WebDriver instance is constructed?

Reply:
Use given one or new instance is created every time.
https://github.com/SeleniumHQ/selenium/blob/6f99c7029876ef69e20b3dee2d6e0e9348ece7f2/py/selenium/webdriver/remote/webdriver.py#L234